### PR TITLE
Port Thruna PDM to VC

### PR DESCRIPTION
**Pull request by liam-ilan**
### Summary
Ported Thruna PDM to VC, and added shutdown loop checks.

### Changelist 
- `efuse`, `lowVoltageBattery`, and `shutdown` io layers added to VC.
- `lowVoltageBattery`, and `shutdown` app layers added to VC.
- Transferred tests from PDM to VC.

### Testing Done
- Software tests pass.
- To be flashed on VC along with Amir's DCM port.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).
**This PR only includes PDM ported code, should be kept in one PR**

### CODE REVIEW

**Comment by liam-ilan** [comment]
Looks like this fixed the md5/Vehicle dynamics errors!

**Comment by liam-ilan** [comment]
For reference, the previous PR, https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/1186

**Review by philam2001 — COMMENTED** [review]
Btw, there were some PDM changes in the vehicle dynamics testing PR, so you might want to port those changes over

File: firmware/quadruna/VC/src/app/states/app_initState.c [hunk 1]
```diff
@@ -12,6 +23,14 @@ static void initStateRunOnTick1Hz(void)
 static void initStateRunOnTick100Hz(void)
 {
     app_allStates_runOnTick100Hz();
+
+    bool is_inverter_on_or_drive_state =
+        app_canRx_BMS_State_get() == BMS_INVERTER_ON_STATE || app_canRx_BMS_State_get() == BMS_DRIVE_STATE;
```
Comment (Jugveer-Sandher) [inline firmware/quadruna/VC/src/app/states/app_initState.c:28]: Need to check if BMS is also in precharge `app_canRx_BMS_State_get() == BMS_PRECHARGE_STATE`

File: firmware/quadruna/VC/src/app/states/app_initState.c [hunk 2]
```diff
@@ -12,6 +23,14 @@ static void initStateRunOnTick1Hz(void)
...
+
+    if (is_inverter_on_or_drive_state)
+    {
+        app_stateMachine_setNextState(app_driveState_get());
```
Comment (Jugveer-Sandher) [inline firmware/quadruna/VC/src/app/states/app_initState.c:32]: You may need another state here so the state transitions for the VC can go from init -> inverter_on -> drive. Where you would have the PDM `is_inverter_on_or_drive_state` check in init and the DCM to drive state checks in inverter_on state

File: firmware/quadruna/VC/src/app/app_powerManager.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    [POWER_MANAGER_SHUTDOWN] = {
+        .efuses = {
+            [EFUSE_CHANNEL_SHDN] = true,
+            [EFUSE_CHANNEL_LV] = true,
+            [EFUSE_CHANNEL_PUMP] = false,
+            [EFUSE_CHANNEL_AUX] = false,
+            [EFUSE_CHANNEL_INV_R] = false,
+            [EFUSE_CHANNEL_INV_L] = false,
+            [EFUSE_CHANNEL_TELEM] = true,
+            [EFUSE_CHANNEL_BUZZER] = true,
```
Comment (Jugveer-Sandher) [inline firmware/quadruna/VC/src/app/app_powerManager.c:21]: This only needs be on during drive

File: firmware/quadruna/VC/src/app/states/app_driveState.c [hunk 4]
```diff
@@ -1,7 +1,15 @@
...
+#include "app_powerManager.h"
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "app_utils.h"
 
-static void driveStateRunOnEntry(void) {}
+static void driveStateRunOnEntry(void)
+{
+    app_canTx_VC_State_set(VC_DRIVE_STATE);
+    app_powerManager_setState(POWER_MANAGER_DRIVE);
```
Comment (Jugveer-Sandher) [inline firmware/quadruna/VC/src/app/states/app_driveState.c:11]: Would need to move this to the new inverter_on state

**Comment by liam-ilan** [comment]
> Btw, there were some PDM changes in the vehicle dynamics testing PR, so you might want to port those changes over

@philam2001 added these in!

File: firmware/quadruna/VC/src/app/states/app_driveState.c (hunk 4 above)
Comment (liam-ilan) [reply firmware/quadruna/VC/src/app/states/app_driveState.c:11]: handled, inverter_on state will be populated in @AmirTajaddodi's PR, for now it remains empty.

File: firmware/quadruna/VC/src/app/states/app_initState.c (hunk 1 above)
Comment (liam-ilan) [reply firmware/quadruna/VC/src/app/states/app_initState.c]: fixed!

File: firmware/quadruna/VC/src/app/states/app_initState.c (hunk 2 above)
Comment (liam-ilan) [reply firmware/quadruna/VC/src/app/states/app_initState.c]: done!

File: firmware/quadruna/VC/src/app/app_powerManager.c (hunk 3 above)
Comment (liam-ilan) [reply firmware/quadruna/VC/src/app/app_powerManager.c]: done!

**Review by gtaharaedmonds — APPROVED** [review]
LGTM

**Pull request merged**